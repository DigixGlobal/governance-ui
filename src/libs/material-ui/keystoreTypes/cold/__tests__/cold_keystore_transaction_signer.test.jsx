import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ColdKeystoreTransactionSigner from '../cold_keystore_transaction_signer';

const txData = {
  to: '0x54341',
  from: '0x543535',
  data: {},
  value: 10,
  nonce: 10,
  gas: 100000,
  gasPrice: 2,
  chainId: 1,
};

describe('<ColdKeystoreTransactionSigner />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <ColdKeystoreTransactionSigner
        txData={txData}
        hideTxSigningModal={jest.fn}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
