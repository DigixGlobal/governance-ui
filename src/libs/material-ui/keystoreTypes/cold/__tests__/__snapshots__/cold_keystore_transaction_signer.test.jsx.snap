// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColdKeystoreTransactionSigner /> renders correctly 1`] = `
<div>
  <Menu
    fluid={true}
    widths={2}
  >
    <MenuItem
      active={true}
      onClick={[Function]}
    >
      1: Sign
    </MenuItem>
    <MenuItem
      active={false}
      onClick={[Function]}
    >
      2: Broadcast
    </MenuItem>
  </Menu>
  <QrCode
    data="{\\"to\\":\\"0x54341\\",\\"from\\":\\"0x543535\\",\\"data\\":{},\\"value\\":10,\\"nonce\\":10,\\"gas\\":100000,\\"gasPrice\\":2,\\"chainId\\":1}"
  />
</div>
`;
